spec.bs

```diff
@@ -2444,6 +2448,28 @@ algorithms to achieve the outcomes described in the above explanatory content.
   /fenced-frame/permission-notification.https.html
 </wpt>
 
+<h3 id=cssom-monkeypatch>CSSOM View</h3>
+
+The [[!CSSOM-VIEW]] specification defines the {{Element/scrollIntoView()}} method that calls the
+[=scroll a target into view=] algorithm. This will not only scroll the {{Element}} or
+[=css2/viewport=] to make the target visible, but will also scroll [=tree/ancestor=]s if necessary
+to make the target visible, essentially causing the scroll to "bubble up". This means that
+{{Element/scrollIntoView()}} performed in a [=child navigable=] or [=fenced navigable container/
+fenced navigable=] can be observed by its embedder, allowing for collusion across a fenced frame
+boundary. This section patches the [=scroll a target into view=] algorithm to prevent that collusion
+at the expense of some utility.
+
+<div algorithm=scroll-target-into-view>
+  Modify the [=scroll a target into view=] algorithm to add a step at the end of the algorithm that
+  reads:
+
+  14. If <var ignore>scrolling box</var>'s associated {{Element}} or [=css2/viewport=]'s associated
```

Comment:
This reads slightly confusing. What you're trying to get at is whether either:
 - scrolling box's associated Element's node navigable's traversable navigable; or
 - scrolling box's associated viewport's associated Document's node navigable's traversable navigable

... is a fenced navigable container. That's the condition you're shooting for right? In that case, I'd strive to be a little more verbose here by being a little redundant as I am above. As it stands, now it ambiguously can read as if you're keying off of whether (a) the associated Element is a fenced navigable container, or (b) the associated viewport's associated Document's node navigable's traversable navigable..... is a fenced navigable container. But that seems wrong, because I don't think the associated element could ever be a fenced navigable container, because the associated element should be outside of the viewport I think (the viewport I think would be everything _inside_ of a fenced navigable container, and we'd only ever have an associated viewport when bubbling out?).

Let me know if that matches your understanding. In any case, I guess we should try and make it is unambiguous as possible, even at the cost of brevity.

Reply:
Yes. My intention is to say "If scrolling box’s associated ([Element](https://dom.spec.whatwg.org/#element) or [viewport](https://www.w3.org/TR/CSS21/visuren.html#viewport))'s associated [Document](https://dom.spec.whatwg.org/#document)'s...". I'll write each case out fully to remove any ambiguity.